File: app/src/main/java/the/mrsmile/foodorderer/adapters/RecyclerAdapter.java
package the.mrsmile.foodorderer.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import the.mrsmile.foodorderer.R;
import the.mrsmile.foodorderer.models.CategoryItems;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.RecyclerViewHolder> {

    List<CategoryItems> list;
    onClick listener;
    public RecyclerAdapter(List<CategoryItems> list,onClick listener) {
        this.list = list;
        this.listener = listener;

    }

    @NonNull
    @Override
    public RecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycler_main, parent, false);
        return new RecyclerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerViewHolder holder, int position) {
        CategoryItems currentItem = list.get(position);
        holder.title.setText(currentItem.getTitle());
        holder.desc.setText(currentItem.getDesc());
        holder.price.setText(currentItem.getPrice());
        holder.imageView.setImageResource(currentItem.getImage());

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class RecyclerViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        ImageView imageView;
        TextView title, desc, price;

        public RecyclerViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.image);
            title = itemView.findViewById(R.id.tvTitleRecyclerMain);
            desc = itemView.findViewById(R.id.tvDescRecyclerMain);
            price = itemView.findViewById(R.id.tvPriceRecyclerMain);
        }

        {
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            listener.onItemClick(getAdapterPosition());
        }
    }

    public interface onClick {
        void onItemClick(int position);
    }
}

File: app/src/main/java/the/mrsmile/foodorderer/adapters/RecyclerAdapter.kt
package the.mrsmile.foodorderer.adapters

import android.content.Context
import android.util.Log
import the.mrsmile.foodorderer.models.RecommendedItems
import androidx.recyclerview.widget.RecyclerView
import the.mrsmile.foodorderer.adapters.RecyclerAdapter.RecyclerViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView import the.mrsmile.foodorderer.R
import android.widget.TextView
import com.bumptech.glide.Glide

class RecyclerAdapter(
    private var list: List<RecommendedItems>,
    var listener: OnRecommendedClick,
    val context: Context
) :
    RecyclerView.Adapter<RecyclerViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.recycler_main, parent, false)
        return RecyclerViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {

        val currentItem = list[position]
        holder.title.text = currentItem.title
        holder.desc.text = currentItem.desc
        holder.price.text = currentItem.price.toString()
        if (!list[position].image.isNullOrEmpty())
        Glide.with(context).load(currentItem.image).placeholder(R.drawable.placeholder2).dontAnimate().centerCrop().into(holder.image)

    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class RecyclerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),
        View.OnClickListener {
        val image: ImageView = itemView.findViewById(R.id.iv_home_ad)
        val title: TextView = itemView.findViewById(R.id.tvTitleRecyclerMain)
        val desc: TextView = itemView.findViewById(R.id.tvDescRecyclerMain)
        val price: TextView = itemView.findViewById(R.id.tvPriceRecyclerMain)

        override fun onClick(view: View) {
            listener.onRecommendedItemClick(absoluteAdapterPosition)
        }

        init {
            itemView.setOnClickListener(this)
        }
    }

    interface OnRecommendedClick {
        fun onRecommendedItemClick(position: Int)
    }
}